### PR TITLE
Adds StandardMessageCodec benchmark

### PR DESCRIPTION
**Pull request by gaaclarke**
This is a benchmark to land before I land my performance fixes.

test exempt: is a test

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Hixie** [comment]
Please don't mark your own PRs as test-exempt, the whole point of the process is that every PR that doesn't have tests needs to be manually checked so as to denormalize the behaviour of writing PRs without tests.

**Comment by gaaclarke** [comment]
> Please don't mark your own PRs as test-exempt, the whole point of the process is that every PR that doesn't have tests needs to be manually checked so as to denormalize the behaviour of writing PRs without tests.

@Hixie everything that is landed is reviewed, so it is manually checked.  In the case of writing tests it seemed we could trust the author and the reviewer to make that decision and that your attention was only required in more nuanced situations.  In any case, I don't think there is harm communicating the explanation.  Let me know if you want to format it differently.

File: dev/benchmarks/microbenchmarks/lib/stocks/build_bench_profiled.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
```
Comment (dnfield) [inline dev/benchmarks/microbenchmarks/lib/stocks/build_bench_profiled.dart:1]: Is this file intended as part of this change?

**Review by dnfield — APPROVED** [review]

**Comment by dnfield** [comment]
I think for this case we should teach the bot about tests under the `dev/benchmarks` folder....

File: dev/benchmarks/microbenchmarks/lib/stocks/build_bench_profiled.dart (hunk 1 above)
Comment (gaaclarke) [reply dev/benchmarks/microbenchmarks/lib/stocks/build_bench_profiled.dart]: Nope, was on the wrong base.  Fixed.

**Comment by Hixie** [comment]
test-exempt: is a test
